crucible-mir/crucible-mir.cabal

```diff
@@ -14,11 +15,106 @@ maintainer:          spernsteiner@galois.com
 copyright:           2017-2023 Galois, Inc.
 category:            Web
 build-type:          Simple
-cabal-version:       2.0
 extra-doc-files:  README.md, CHANGELOG.md
 
-library
+common shared
+  -- Specifying -Wall and -Werror can cause the project to fail to build on
+  -- newer versions of GHC simply due to new warnings being added to -Wall. To
+  -- prevent this from happening we manually list which warnings should be
+  -- considered errors. We also list some warnings that are not in -Wall, though
+  -- try to avoid "opinionated" warnings (though this judgement is clearly
+  -- subjective).
+  --
+  -- Warnings are grouped by the GHC version that introduced them, and then
+  -- alphabetically.
+  --
+  -- A list of warnings and the GHC version in which they were introduced is
+  -- available here:
+  -- https://ghc.gitlab.haskell.org/ghc/doc/users_guide/using-warnings.html
+
+  -- Since GHC 9.6 or earlier:
+  ghc-options:
+    -Wall
+    -Werror=ambiguous-fields
+    -Werror=compat-unqualified-imports
+    -Werror=deferred-type-errors
```

Comment:
Is there a way to make these error only in CI?  `-Werror` can be quite annoying during development, since it forces you to fix trivial issues like unused variables while you're still trying to figure out whether your proposed solution actually works.